### PR TITLE
Fix #851 - Remove empty non-root groups automatically

### LINKED ISSUE #851: Groups and Applications may not have the same identifier
I created app `/something/app`, but then I decided that I'd better have `/something/app/service`. I removed `/something/app`, but when I try to create `/something/app/service`, I get an error:

```
Groups and Applications may not have the same identifier
```

I think there should be no error here.

_UPDATE_

The steps to reproduce this problem are:

1. `PUT /v2/apps/something/app`
2. `DELETE /v2/apps/something/app`
3. `PUT /v2/apps/something`, not `PUT /v2/apps/something/app/service`

### PR DESCRIPTION
**Pull request by kolloch**
Remove groups automatically which do not contain apps or groups anymore.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/state/Group.scala [hunk 1]
```diff
@@ -7,6 +7,7 @@ import mesosphere.marathon.state.PathId._
 import org.jgrapht.DirectedGraph
 import org.jgrapht.alg.CycleDetector
 import org.jgrapht.graph._
+import org.slf4j.LoggerFactory
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/state/Group.scala]: left over?

**Comment by aquamatthias** [comment]
:+1:

**Comment by kolloch** [comment]
Thanks for the quick review! Was your review complete?

**Comment by drexin** [comment]
LGTM

**Pull request merged**